src/Sluggable/SluggableListener.php

```diff
@@ -440,11 +440,11 @@ private function generateSlug(SluggableAdapter $ea, object $object): void
                 }
 
                 // cut slug if exceeded in length
-                if (isset($mapping['length']) && strlen($slug) > $mapping['length']) {
-                    $slug = substr($slug, 0, $mapping['length']);
+                if (isset($mapping['length']) && strlen($slug) > ($mapping->length ?? $mapping['length'])) {
+                    $slug = substr($slug, 0, $mapping->length ?? $mapping['length']);
```

Comment:
These issets in this file look like they might cause a problem.  Might be better to just extract these to variables.

```suggestion
                $length = $mapping->length ?? $mapping['length'] ?? null;
                if (null !== $length && strlen($slug) > $length) {
                    $slug = substr($slug, 0, $length);
```